crates/sargon/src/gateway_api/methods/page_methods.rs

```diff
@@ -0,0 +1,160 @@
+use crate::prelude::*;
+use std::future::Future;
+
+impl GatewayClient {
+    /// Load all pages of a paginated API call that returns a `PageResponse`.
+    /// Parameters:
+    /// - `api_call`: A function that takes an optional cursor and returns a future executing the
+    /// corresponding API call.
+    ///
+    /// Returns: A collection of the items from all pages.
+    pub async fn load_all_pages<T, F, Fut>(&self, api_call: F) -> Result<Vec<T>>
+    where
+        F: Fn(Option<String>) -> Fut,
+        Fut: Future<Output = Result<PageResponse<T>>>,
+    {
+        let mut items: Vec<T> = Vec::new();
+        let mut more_to_load = true;
+        let mut cursor: Option<String> = None;
+        while more_to_load {
+            let response = api_call(cursor.clone()).await?;
+            items.extend(response.items);
+            cursor = response.next_cursor;
+            more_to_load = cursor.is_some();
+        }
+
+        Ok(items)
+    }
+
+    /// Load all pages of a paginated API call that returns a `PageResponse`.
+    /// Parameters:
+    /// - `account_address`: The address of the account to load pages for.
+    /// - `api_call`: A function that takes an `AccountPageRequest` and returns a future executing
+    /// the corresponding API call.
+    ///
+    /// Returns: A collection of the items from all pages.
+    pub async fn load_all_account_pages<T, F, Fut>(
+        &self,
+        account_address: AccountAddress,
```

Comment:
Along with account address we need to pass in the ledger state if the account entity, so preferences pagination uses the same ledger state as the account entity. The ledger state will need to be passed in by the host.